source/Halibut/ServiceModel/IPendingRequestQueue.cs

```diff
@@ -7,6 +8,6 @@ public interface IPendingRequestQueue
     {
         bool IsEmpty { get; }
         void ApplyResponse(ResponseMessage response);
-        RequestMessage Dequeue();
+        Task<RequestMessage> Dequeue();
```

Comment:
Ah, I see that is a public API, but OD doesn't use it. Sure can make it additive.